go/ct/spc/specification_map.go

```diff
@@ -59,11 +59,10 @@ func (s *specificationMap) GetRulesFor(state *st.State) []Rule {
 	var opString string
 	if err != nil {
 		opString = "noOp"
+	} else if state.Revision == common.R99_UnknownNextRevision || state.Status != st.Running {
+		opString = "noOp"
 	} else {
 		opString = op.String()
-		if state.Revision == common.R99_UnknownNextRevision || state.Status != st.Running {
-			getRules("noOp")
-		}
 	}
 
 	getRules(opString)
```

Comment:
Since `getRules` is now only used once, please also inline the code for this function here. It is very unusual to have a function like this having the "side-effect" of building up a result.

Reply:
Right, I inlined the lambda function. 
There is the `TestSpecificationMap_SameRulesPerOperation` unit test which checks that for random states the map returns all applicable rules.